app/models/occupation.rb

```diff
@@ -14,22 +14,50 @@ class Occupation < ApplicationRecord
   number_of_shards = Rails.env.production? ? 2 : 1
   es_settings = {
     index: {
-      number_of_shards: number_of_shards
+      number_of_shards: number_of_shards,
+      analysis: {
+        tokenizer: {
+          autocomplete_tokenizer: {
+            type: "edge_ngram",
+            min_gram: 2,
+            max_gram: 20,
+            token_chars: ["letter", "digit", "punctuation"]
+          }
+        },
+        analyzer: {
+          autocomplete: {
+            tokenizer: "autocomplete_tokenizer",
+            filter: ["lowercase"]
+          },
+          autocomplete_search: {
+            tokenizer: "standard",
+            filter: ["lowercase"]
+          }
+        }
+      }
     }
   }
 
   settings(es_settings) do
     mappings dynamic: false do
-      indexes :title, type: :text, analyzer: :english
+      indexes :title, type: :text, analyzer: :autocomplete, search_analyzer: :autocomplete_search
+      indexes :rapids_code, type: :text, analyzer: :autocomplete, search_analyzer: :autocomplete_search
+      indexes :onet_code, type: :text, analyzer: :autocomplete, search_analyzer: :autocomplete_search
     end
   end
 
   def as_indexed_json(_ = {})
     as_json(
-      only: [:title]
+      only: [:title, :rapids_code, :onet_code]
```

Comment:
```suggestion
      only: [:title, :rapids_code]
```
This top section is only for fields on the model.